backend/src/main/java/com/htecgroup/skynest/annotation/OnlyManagerCanBeDemoted.java

```diff
@@ -0,0 +1,21 @@
+package com.htecgroup.skynest.annotation;
+
+import com.htecgroup.skynest.annotation.validator.DemoteValidator;
+
+import javax.validation.Constraint;
+import javax.validation.Payload;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Retention(RetentionPolicy.RUNTIME)
+@Target({ElementType.PARAMETER})
+@Constraint(validatedBy = DemoteValidator.class)
+public @interface OnlyManagerCanBeDemoted {
```

Comment:
If you have a annotation that is checking if user is manager, use that.
If not since I am not sure if you have different annotations for checking if the user is admin, worker or manager.. you can combine that in one annotation and use it with different input params.
Try to make the annotation reusable.. check with @MarkoRadunovicHtec , I already showed him how to do it

Reply:
Marko helped me, and I think we did it.